internal/controller/environment/controller.go

```diff
@@ -73,24 +75,44 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, err
 	}
 
-	previousCondition := meta.FindStatusCondition(environment.Status.Conditions, controller.TypeAvailable)
-
-	environment.Status.ObservedGeneration = environment.Generation
-	if err := controller.UpdateCondition(
-		ctx,
-		r.Status(),
-		environment,
-		&environment.Status.Conditions,
-		controller.TypeAvailable,
-		metav1.ConditionTrue,
-		"EnvironmentAvailable",
-		"Environment is available",
-	); err != nil {
-		return ctrl.Result{}, err
-	} else {
-		if previousCondition == nil {
-			r.Recorder.Event(environment, corev1.EventTypeNormal, "ReconcileComplete", "Successfully created "+environment.Name)
+	old := environment.DeepCopy()
+
+	_, err := r.makeEnvironmentContext(ctx, environment)
```

Comment:
Why do we need to create the context here? It seems, we are not using it.

Reply:
basically, the makeEnvironmentContext acts here as a kind of a validation step, it is checking the error of this to make that validation. 
for example if the dataplane is not found for the corresponding environment, it throws an error with an k8s event.

Reply:
We are already calling this inside finalize too, right? Do you think we can't use this logic there? 
Even if we validate here, the state can be changed when it reaches the actual finalize logic as this is not fully depend on the cache

Reply:
yeah, +1 to do the validation in there too..

Reply:
As per the offline discussion we had, having this here doesn't make sense as we don't have any resource provisioning with the environment reconciliation. It is just checking whether the corresponding dataplane exists. If there is an environment created with non existing dataplane, this could trigger up lot of events, and it could become entries in etcd eventually. So removed this from here and only have this in the finalize function.